wys/api/python/wys_api.py

```diff
@@ -0,0 +1,351 @@
+                  api_id integer NOT NULL,
+                  address text,
+                  sign_name text,
+                  dir text,
+                  start_date date);
+            """
+            cur.execute(str(string))
+            for signs_iterator in signs_iterator:
+                location=signs_iterator[0]
+                statistics=get_statistics(location,start_date,api_key)
+                loc_info=statistics['LocInfo']
+                loc=loc_info['Location']
+                lat=loc['geocode'][0]
+                long=loc['geocode'][1]
+                address=loc['address']
+                name=loc['name']
+                if 'SB' in name:
+                    direction='SB'
+                elif 'NB' in name:
+                    direction='NB'
+                elif 'WB' in name:
+                    direction='WB'
+                elif 'EB' in name:
+                    direction='EB'
+                else:
+                    direction=None
+                temp=[location, address, name, direction, start_date, lat, long]
+                loc_table.append(temp)
+                signs_iterator=signs_list
+                
+            execute_values(cur, 'INSERT INTO daily_intersections (api_id, address, sign_name, dir, start_date, lat, lng) VALUES %s', loc_table)
+            string="""
+                        WITH locations AS (
+                        SELECT api_id, address, sign_name, dir, lat, lng, max(start_date)
+                        FROM wys.locations 
+                        GROUP BY api_id, address, sign_name, lat, lng, dir),
+                        
+                        
+                        
+                        differences AS (
+            
+                        SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, lat, lng FROM daily_intersections A
+                        LEFT JOIN locations B ON A.api_id = B.api_id
+                        AND A.sign_name = B.sign_name 
+                        WHERE B.sign_name IS NULL 
+                        
+                        UNION
+                        
+                        
+                        SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, lat, lng FROM daily_intersections A
+                        LEFT JOIN locations B ON A.api_id = B.api_id
+                        AND B.address = A.address
+                        WHERE B.address IS NULL
+                        
+                        UNION 
+                        
+                        SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, lat, lng FROM daily_intersections A
+                        LEFT JOIN locations B ON A.api_id = B.api_id
+                        AND B.lat = A.lat
+                        WHERE B.lat IS NULL
+                        
+                        SELECT a.api_id, a.address, a.sign_name, a.dir, start_date, lat, lng FROM daily_intersections A
```

Comment:
This seems duplicated and also there's no UNION between this SELECT and the one before it.